crates/cli/src/subcommands/generate/mod.rs

```diff
@@ -132,7 +132,12 @@ pub async fn exec(config: Config, args: &clap::ArgMatches) -> anyhow::Result<()>
         } else {
             build::exec_with_argstring(config.clone(), project_path, build_options).await?
         };
-        extract_descriptions(&wasm_path)?
+        let spinner = indicatif::ProgressBar::new_spinner();
+        spinner.enable_steady_tick(60);
+        spinner.set_message("Compiling wasm...");
+        let module = compile_wasm(&wasm_path)?;
+        spinner.set_message("Extracting schema from wasm...");
+        extract_descriptions_from_module(module)?
```

Comment:
my main nit is that it seems like there are several places where we take a long time to compile something, and I'm confused why we're adding a spinner for this specific one. Like, it's an improvement, but it's also a bit confusing to not be doing this everywhere if we're doing it (and potentially factoring out some common helpers etc.)

Reply:
Which other places did you have in mind?

Reply:
I mean, actual `build` is the main one that comes to mind that I feel like people have mentioned before.